src/Service/NotificationReceiverService.php

```diff
@@ -114,35 +115,16 @@ public function process(Request $requestObject)
             throw new AuthorizationException();
         }
 
-        // Is the plugin configured to live environment
-        $pluginMode = $this->configurationService->getEnvironment();
+        // Is the plugin configured to live environment?
+        $pluginIsLive = $this->configurationService->getEnvironment() === Environment::LIVE;
 
-        // Validate notification and process the notification items
-        if (!empty($request['live']) && $this->validateNotificationMode($request['live'], $pluginMode)) {
-            $acceptedMessage = '[accepted]';
-
-            // Process each notification item
-            foreach ($request['notificationItems'] as $notificationItem) {
-                $notificationItem['NotificationRequestItem']['live'] = $request['live'];
-                if (!$this->processNotificationItem($notificationItem['NotificationRequestItem'], $salesChannelId)) {
-                    throw new ValidationException();
-                }
-            }
-
-            // Run the query for checking unprocessed notifications, do this only for test notifications coming from
-            // the Adyen Customer Area
-            if ($isTestNotification) {
-                $unprocessedNotifications = $this->notificationService->getNumberOfUnprocessedNotifications();
-                if ($unprocessedNotifications > 0) {
-                    $acceptedMessage .= "\nYou have $unprocessedNotifications unprocessed notifications.";
-                }
-            }
-            $this->logger->info('The result is accepted');
+        // Is the notification coming from live environment?
+        if(!is_bool($request['live'] ?? null)) {
+            $request['live'] = isset($request['live']) && $request['live'] === 'true';
+        }
 
-            return new JsonResponse(
-                $this->returnAccepted($acceptedMessage)
-            );
-        } else {
+        // Checks if notification mode and the store mode configuration matches
+        if ($request['live'] !== $pluginIsLive) {
```

Comment:
Maybe we don't need to reject notifications if the modes don't match. What do you think @cyattilakiss?

Reply:
@acampos1916 We can receive and store those notifications - we should not do anything else other than the security checks before storing them in the db. 
When we process them though we should log an error and stop processing the notification which has the mismatch.

Reply:
Okay, I will remove this particular line of validation

Reply:
Gotcha, so yes @peterojo, we can remove the mode check from the receiving service